src/persist/clients.ts

```diff
@@ -131,12 +131,15 @@ export async function assertClientExists(
   id: sync.ClientID,
   dagRead: dag.Read,
 ): Promise<void> {
-  const client = await getClient(id, dagRead);
-  if (!client) {
+  if (await isClientMissing(id, dagRead)) {
```

Comment:
Maybe we should just have persist return a boolean instead of throwing the error?